go/keymanager/client/client.go

```diff
@@ -80,14 +81,26 @@ func (c *Client) CallRemote(ctx context.Context, data []byte) ([]byte, error) {
 			Endpoint:  api.EnclaveRPCEndpoint,
 			Payload:   data,
 		})
-		if status.Code(err) == codes.PermissionDenied {
+		switch {
+		case err == nil:
+		case status.Code(err) == codes.PermissionDenied:
 			// Calls can fail around epoch transitions, as the access policy
 			// is being updated, so we must retry.
 			return err
+		case status.Code(err) == codes.Unavailable:
+			// XXX: HACK: Find a better way to determine the root cause.
+			if strings.Contains(err.Error(), "tls: bad public key") {
```

Comment:
Unfortunately, due to gRPC's use of wrapping RPC errors (see [here](https://github.com/grpc/grpc-go/blob/754ee590a4f386d0910d887f3b8776354042260b/rpc_util.go#L782)) it is not possible to get the original error.